airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcPartitionFactory.kt

```diff
@@ -350,15 +352,30 @@ class MysqlJdbcPartitionFactory(
                     }
                     LeafAirbyteSchemaType.TIMESTAMP_WITH_TIMEZONE -> {
                         val timestampInStatePattern = "yyyy-MM-dd'T'HH:mm:ss"
+                        val formatter =
+                            DateTimeFormatterBuilder()
+                                .appendPattern(timestampInStatePattern)
+                                .optionalStart()
+                                .appendFraction(ChronoField.NANO_OF_SECOND, 1, 9, true)
+                                .optionalEnd()
+                                .optionalStart()
+                                .optionalStart()
+                                .appendLiteral(' ')
+                                .optionalEnd()
+                                .appendOffset("+HH:mm", "Z")
+                                .optionalEnd()
+                                .toFormatter()
+
                         try {
-                            val formatter: DateTimeFormatter =
-                                DateTimeFormatter.ofPattern(timestampInStatePattern)
-                            Jsons.valueToTree(
-                                LocalDateTime.parse(stateValue, formatter)
-                                    .minusDays(1)
-                                    .atOffset(java.time.ZoneOffset.UTC)
-                                    .format(OffsetDateTimeCodec.formatter)
-                            )
+                            val offsetDateTime =
+                                try {
+                                    OffsetDateTime.parse(stateValue, formatter)
+                                } catch (_: DateTimeParseException) {
```

Comment:
If value include no `+xx:xx` or `Z` then we create an offset datetime with an assumed UTC offset.
This is done via a LocalDate.